p2p/src/net/default_backend/backend.rs

```diff
@@ -57,10 +57,15 @@ use super::{
     types::{HandshakeNonce, Message, P2pTimestamp},
 };
 
-/// Buffer size of the channel to the SyncManager peer task.
-/// How many unprocessed messages can be sent before the peer's event loop is blocked.
-// TODO: Decide what the optimal value is (for example, by comparing the initial block download time)
-const SYNC_CHAN_BUF_SIZE: usize = 20;
+/// Buffer sizes for the channels used by Peer to send peer messages to other parts of p2p.
+///
+/// If the number of unprocessed messages exceeds this limit, the peer's event loop will be
+/// blocked; this is needed to prevent DoS attacks where a peer would overload the node with
+/// requests, which may lead to memory exhaustion.
+/// Note: the value was chosen pretty much arbitrarily, but judging by the initial block download
+/// time there is no difference between 20 and 10000, so the former seems reasonable enough.
+const SYNC_MSG_CHAN_BUF_SIZE: usize = 20;
```

Comment:
I really think it should much, much bigger. I mean, we went from infinity to 20? I wouldn't mind going for 1000 or even more.

Reply:
The sync messages can indeed vary wildly, from a single `Id` to a whole `Block`, which can be up to 1Mb in size. We've discussed this with Sam on Friday and decided that 100 should be a reasonable limit for the sync messages channel.
Peer manager messages, on the other hand, are relatively small and a bigger limit can be chosen, so we've decided on 1000. (I personally think that it's not necessary, but it won't hurt because those messages are small).